### PR TITLE
Mpiio test

### PR DESCRIPTION
**Pull request by hirschsn**
Fixes missing MPI-IO test (see #2071).

Description of changes:
 - Add a test that creates random particles, dumps them, reads them in, again and verifies the read particles.

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

**Comment by hirschsn** [comment]
Oops, some of the test pipelines fail. I'll look into it. :)

**Comment by hirschsn** [comment]
Should be good now.

File: testsuite/python/mpiio.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    again and then checks the input against the initially created random
+    particles.
+    """
+    s = espressomd.system.System(box_l=[1, 1, 1])
+    # Just a bunch of random interaction such that add_bond does not throw
+    for i in range(nbonds):
+        s.bonded_inter[i] = AngleHarmonic(bend=i, phi0=i)
+    test_particles = random_particles()
+
+    def setup_sample_system(self):
```
Comment (KaiSzuttor) [inline testsuite/python/mpiio.py]: could you please use the python unittest class' special functions like ```setUp``` etc. for preparing and cleanup?

File: testsuite/python/mpiio.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        """Sets up a system from test_particles."""
+        for p in self.test_particles:
+            self.s.part.add(id=p.id, type=p.type, pos=p.pos, v=p.v)
+            for b in p.bonds:
+                self.s.part[p.id].add_bond(b)
+
+    def assertAllEqual(self, v, w):
```
Comment (KaiSzuttor) [inline testsuite/python/mpiio.py]: please check if the numpy testing module provides this functionality

**Review by fweik — DISMISSED** [review]

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]
thanks for your effort! could you please address my comments

**Comment by hirschsn** [comment]
@KaiSzuttor, thanks for the comments, done.

**Review by KaiSzuttor — APPROVED** [review]

**Comment by KaiSzuttor** [comment]
thanks steffen

**Pull request merged**